creator-node/src/routes/files.js

```diff
@@ -484,6 +486,50 @@ module.exports = function (app) {
    */
   app.get('/ipfs/:dirCID/:filename', getDirCID)
 
+  /**
+   * Serves information on existence of given cids
+   * @param req
+   * @param req.body
+   * @param {string} req.body.cids the cids to check existence for
+   * @dev This route can have a large number of CIDs as input, therefore we use a POST request.
+   * TODO: Remove support for directories
+   */
+  app.post('/batch_cids_exist', handleResponse(async (req, res) => {
+    const { cids } = req.body
+
+    if (cids.length > BATCH_CID_ROUTE_LIMIT) {
+      return errorResponseBadRequest(`Too many CIDs passed in, limit is ${BATCH_CID_ROUTE_LIMIT}`)
+    }
+
+    const queryResults = (await models.File.findAll({
+      attributes: ['multihash', 'storagePath'],
+      where: {
+        multihash: {
+          [models.Sequelize.Op.in]: cids
```

Comment:
you can add the raw: true property here to make this more performant. the ORM gives you an instance of a FIle model object that you can perform db operations on, but raw says just give me the json data without the extra stuff which is a lot more lightweight in heavy routes like this

Reply:
@cheran-senthil bump on this one. here's an example https://github.com/AudiusProject/audius-protocol/blob/master/creator-node/src/routes/nodeSync.js#L58